.gitmodules

```diff
@@ -1,3 +1,3 @@
 [submodule "tools/clang"]
 	path = tools/clang
-	url = ../../wjakob/clang-cindex-python3.git
+	url = git@github.com:wjakob/clang-cindex-python3.git
```

Comment:
Shouldn't that be `https://`, for users that don't have SSH set up?

Reply:
This breaks things. It needs to stay relative. I believe this has come up before, and been refused. If you force SSH or HTTPS, then if you are on a system that doesn't have access to the other, it breaks. Relative makes them work.

I think the best fix, however, would to to add the extra tools via CMake instead of via submodule. I need to understand what these are used for before attempting anything, though.

Reply:
> then if you are on a system that doesn't have access to the other

Would you ever have access to ssh but not to https?

Reply:
Depends on how you have your networking setup. I've had that happen in CI at least; Git complained that everything was not ssh. Clusters might have ssh without https, but not sure about that. But in general, if you want to use SHH, that should not also be grabbing things through https, in general.

The traditional submodule solution might be to move the repo into the pybind org, then someone cloning pybind11 could also clone this repo too, even if they were on gitlab, if they add the submodules.

If I knew what this submodule was for, that would help me provide a better recommendation.

Reply:
Right, OK, yes.

It's used in https://github.com/pybind/pybind11/blob/master/tools/mkdoc.py, btw, as far as I know.

Reply:
@wjakob explains the use in #807.

Reply:
Yep, I've literally written this before because I didn't know it existed. Though I wrote it in CMake instead of Python: https://github.com/GooFit/GooFit/blob/master/python/make_docs_header.cmake

Reply:
PS: Tests currently pass because a) no tests do a recursive checkout and b) this is completely untested.